### PR TITLE
Add RepositoryResource

### PR DESCRIPTION
**Pull request by adjackura**
Still need to add tests for APT GPG key setup, this should be a feature match for current GuestPolicies.

### CODE REVIEW

File: config/repository_resource.go [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+	case *agentendpointpb.ApplyConfigTask_Config_Resource_RepositoryResource_Yum:
+		if !packages.YumExists {
+			return nil, errors.New("cannot manage yum repository because yum does not exist on the system")
+		}
+		r.managedRepository.Yum = &YumRepository{RepositoryResource: r.GetYum()}
+		r.managedRepository.RepoFileContents = yumRepoContents(r.GetYum())
+		filePath = filepath.Join(agentconfig.YumRepoDir(), "osconfig_managed_%s.repo")
+
+	case *agentendpointpb.ApplyConfigTask_Config_Resource_RepositoryResource_Zypper:
+		if !packages.YumExists {
```
Comment (manjunathganga) [inline config/repository_resource.go:263]: ZypperExists?

File: config/repository_resource.go [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+}
+
+func (r *repositoryResource) checkState(ctx context.Context) (inDesiredState bool, err error) {
+	// Check APT gpg key if applicable.
+	if r.managedRepository.Apt != nil && r.managedRepository.Apt.GpgFileContents != nil {
+		match, err := contentsMatch(r.managedRepository.Apt.GpgFilePath, r.managedRepository.Apt.GpgChecksum)
+		if !match {
+			return false, nil
+		}
+		if err != nil {
```
Comment (manjunathganga) [inline config/repository_resource.go:298]: Should this error check happen before the check for match as an error also would return a mismatch?

File: config/repository_resource.go (hunk 1 above)
Comment (adjackura) [reply config/repository_resource.go]: Fixed

File: config/repository_resource.go (hunk 2 above)
Comment (adjackura) [reply config/repository_resource.go:295]: yep, fixed

File: config/repository_resource.go [hunk 3]
```diff
@@ -260,7 +260,7 @@ func (r *repositoryResource) validate() (*ManagedResources, error) {
 		filePath = filepath.Join(agentconfig.YumRepoDir(), "osconfig_managed_%s.repo")
 
 	case *agentendpointpb.ApplyConfigTask_Config_Resource_RepositoryResource_Zypper:
-		if !packages.YumExists {
+		if !packages.ZypperExists {
 			return nil, errors.New("cannot manage yum repository because yum does not exist on the system")
```
Comment (manjunathganga) [inline config/repository_resource.go:264]: The error message also needs to be updated :-)

**Review by manjunathganga — APPROVED** [review]

**Pull request merged**